src/ddmd/expression.d

```diff
@@ -15509,6 +15523,11 @@ extern (C++) final class CmpExp : BinExp
         }
 
         //printf("CmpExp: %s, type = %s\n", e.toChars(), e.type.toChars());
+        if (arrayLowering)
+        {
+            arrayLowering = arrayLowering.semantic(sc);
+            return arrayLowering;
```

Comment:
This style in dmd comes from Walter wanting to be able to trivially insert prints before the return, and I tend to agree.